### PR TITLE
feat(new-trace): Adding performance setup checklist for only error tr…

### PR DESCRIPTION
**Pull request by Abdkhan14**
Org to test: [link](https://perfquotaorg.dev.getsentry.net:7999/issues/?project=4506836555595776&referrer=sidebar&statsPeriod=14d) navigate to the trace for issue details. 

- We now push projects that are part of the trace to the top of the checklist.
- Note: Will improve file-structure of the the new trace quality banners as we add more. 

Screenshot 2024-07-24 at 4 40 35 PM

### CODE REVIEW

File: static/app/views/performance/newTraceDetails/traceWarnings/performanceSetupWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
+import {useEffect, useMemo} from 'react';
```
Comment (Abdkhan14) [inline static/app/views/performance/newTraceDetails/traceWarnings/performanceSetupWarning.tsx:1]: Reused most of the code from old banner: https://github.com/getsentry/sentry/blob/master/static/app/views/performance/traceDetails/content.tsx#L428

File: static/app/components/performanceOnboarding/sidebar.tsx [hunk 2]
```diff
@@ -45,6 +58,12 @@ function PerformanceOnboardingSidebar(props: CommonSidebarProps) {
   const {projectsWithoutFirstTransactionEvent, projectsForOnboarding} =
     filterProjects(projects);
 
+  const priorityProjectIds: Set<string> | null = useMemo(() => {
+    const queryParams = qs.parse(location.search);
+    const decodedProjectIds = decodeProjectIds(queryParams.project);
```
Comment (DominikB2014) [inline static/app/components/performanceOnboarding/sidebar.tsx:63]: I think the `useLocation` hook is preferred here if possible rather then the global location object. In general id use react api's over browser apis.

you can add `const useLocation = location<{project: string | string[]}>`

**Comment by DominikB2014** [comment]
I'm super familiar with the checklist, do you giving me more details on what the old behaviour was, and how it compares with the new behaviour.

File: static/app/components/performanceOnboarding/sidebar.tsx (hunk 2 above)
Comment (Abdkhan14) [reply static/app/components/performanceOnboarding/sidebar.tsx:63]: Yeah I can switch to `useLocation` here

**Review by DominikB2014 — APPROVED** [review]

**Pull request merged**